common/src/main/scala/org/apache/celeborn/common/protocol/message/ControlMessages.scala

```diff
@@ -1079,7 +1083,8 @@ object ControlMessages extends Logging {
             pbHeartbeatFromApplication.getNeedCheckedWorkerListList.asScala
               .map(PbSerDeUtils.fromPbWorkerInfo).toList.asJava),
           pbHeartbeatFromApplication.getRequestId,
-          pbHeartbeatFromApplication.getShouldResponse)
+          pbHeartbeatFromApplication.getShouldResponse,
+          PbSerDeUtils.fromPbUserIdentifier(pbHeartbeatFromApplication.getUserIdentifier))
```

Comment:
user identifier can be null in case of an old client right ? While can result in NPE in `fromPbUserIdentifier` ?

Reply:
thanks, addressed